### PR TITLE
Report json decoding error

### LINKED ISSUE #2107: Improve json decoding error reporting
If you submit invalid json to the API you get a 500 with a message "unable to decode request body". However, the logs are much clearer on this:

```
2020-05-07 12:40:48,270 ERROR    inmanta.protocol.rest.server An exception occured
Traceback (most recent call last):
  File "/opt/inmanta/lib64/python3.6/site-packages/inmanta/protocol/rest/server.py", line 118, in _call
    message = self._transport._decode(self.request.body)
  File "/opt/inmanta/lib64/python3.6/site-packages/inmanta/protocol/rest/__init__.py", line 375, in _decode
    result = cast(JsonType, json.loads(escape.to_basestring(body)))
  File "/usr/lib64/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python3.6/json/decoder.py", line 355, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 5 column 9 (char 131)
```

We can at least disclose that it was a json decoding issue. Not sure if it is safe to report the detailed message that details where exactly the error is. Currently I see no issue for that, because many users of the API do not have access to the server logs.

### PR DESCRIPTION
**Pull request by andraskvr**
# Description

closes #2107

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [x] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: src/inmanta/protocol/rest/server.py [hunk 1]
```diff
@@ -140,6 +141,11 @@ async def _call(self, kwargs: Dict[str, str], http_method: str, call_config: com
                 else:
                     raise exceptions.UnauthorizedException("Access to this resource is unauthorized.")
 
+            except JSONDecodeError:
+                error_message = "The request body couldn't be decoded as a JSON"
```
Comment (bartv) [inline src/inmanta/protocol/rest/server.py:145]: The exception also contains more details about where the errors. Should we provide that? See the original ticket for that. @wouterdb I do not see any security issues to report this.

Comment (wouterdb) [reply src/inmanta/protocol/rest/server.py:145]: agreed

File: src/inmanta/protocol/rest/server.py [hunk 2]
```diff
@@ -140,6 +141,11 @@ async def _call(self, kwargs: Dict[str, str], http_method: str, call_config: com
                 else:
                     raise exceptions.UnauthorizedException("Access to this resource is unauthorized.")
 
+            except JSONDecodeError as e:
+                error_message = f"The request body couldn't be decoded as a JSON: {e}"
+                LOGGER.exception(error_message)
```
Comment (wouterdb) [inline src/inmanta/protocol/rest/server.py:146]: I think this has to be logged on a lower log level than error, as it is not a server error (could be info or even debug)

File: tests/test_protocol.py [hunk 3]
```diff
@@ -1499,3 +1499,18 @@ async def test_required_header_not_present(server):
...
+        Tests sending malformed json to the server
+    """
+    port = opt.get_bind_port()
+    url = f"http://localhost:{port}/api/v2/environment"
+
+    request = HTTPRequest(url=url, method="PUT", body='{"name": env}')
+    client = AsyncHTTPClient()
+    response = await client.fetch(request, raise_error=False)
+    assert response.code == 400
+    assert json.loads(response.body)["message"].startswith("The request body couldn't be decoded as a JSON")
```
Comment (wouterdb) [inline tests/test_protocol.py:1516]: Also add the extended message to the testcase

**Pull request merged**